src/cart/crt_iv.c

```diff
@@ -2756,10 +2756,12 @@ handle_response_cb(const struct crt_cb_info *cb_info)
 		info->cci_rc = cb_info->cci_rc;
 		info->cci_arg = cb_info->cci_arg;
 
-		/* handle timeouts on IV_FETCH synchronously to ensure unregister of bulk buffer
-		 * will occur before freeing it, in case emiter will finally make it
+		/* handle locally generated errors on IV_FETCH synchronously to ensure unregister
+		 * of bulk buffer will occur before freeing it, in case peer will finally make it
 		 */
-		if (cb_info->cci_rc == DER_TIMEDOUT && cb_info->cci_rpc->cr_opc == CRT_OPC_IV_FETCH)
+		if ((cb_info->cci_rc == -DER_TIMEDOUT || cb_info->cci_rc == -DER_EXCLUDED ||
+		     cb_info->cci_rc == -DER_CANCELED) &&
+		    cb_info->cci_rpc->cr_opc == CRT_OPC_IV_FETCH)
```

Comment:
looking at lines 2772-2774 where another error is handled -- do we need to do the similar here?
are we going to leak 'info' otherwise?

Reply:
Well you are probably right, and I should also need to RPC_DECREF(rpc_priv) ...
Why not moving my code to the very beginning of handle_response_cb() as it only relies on cb_info content ?!